### ISSUE TITLE
how to select the best performed model on validation set?

### ISSUE THREAD

**Issue by InternetMedical** [dmis-lab/biobert#94]
Dear authors
Your biobert is an excellent work. Can I ask you a question?
You mentioned that "The performance of your bert-based model usually converges at more than 50 epochs."
And in your run_ner.py line 87:

flags.DEFINE_integer("save_checkpoints_steps", 1000,
                     "How often to save the model checkpoint.")

It seems you do not save the best performed model based on the scores on validation set.
Can you tell me how to save the best performed one ?
How can you judge whether the model is fully trained ?

**Comment by jhyuklee** [comment]
You have to test your model on the validation set, and find the hyperparameters that work the best (including the number of epochs). Then, re-train your model with the (train+dev) set. Hyperparameter of each dataset varies a lot (especially the learning rate and the number of epoch) but it will be safe to choose low learning rate and run them more than 50~100 epochs and see how the performance changes. Thanks.

**Comment by InternetMedical** [comment]
Thanks for your quick reply. 
In other words, the performance of the biobert model converges at about 50th epoch on your dataset. Later, the model might suffer from overfitting. You obtained such phenomenon after running the model on 100 epochs.  Is it right?

**Comment by jhyuklee** [comment]
Yes, but it depends on the dataset you train on.
